### PR TITLE
Add xarray open kwargs to cesm timeseries and cesm history

### LINKED ISSUE #80: Add ability to add zarr backend for parsers
Add `xarray` arguments as optional in the `parse_cesm_timeseries` and `parse_cesm_history` parsers, to enable reading from other formats (ex. `netCDF`, `zarr`, `grib`)

### PR DESCRIPTION
**Pull request by mgrover1**
Closes #80 - allows user to specify an engine for reading in the data into `xarray`

### CODE REVIEW

File: ecgtools/parsers/cesm.py [hunk 1]
```diff
@@ -99,7 +99,7 @@ def _join(a):
     return date
 
 
-def parse_cesm_history(file, user_streams_dict={}):
+def parse_cesm_history(file, user_streams_dict={}, xarray_open_engine='netcdf4'):
```
Comment (andersy005) [inline ecgtools/parsers/cesm.py:102]: Could we make this more generic to allow passing additional arguments?

```suggestion
def parse_cesm_history(file, user_streams_dict={}, xarray_open_kwargs=None):
  _default_kwargs = {'engine': 'netcdf4', chunks: {}, decode_times=False}
  if xarray_open_kwargs is None:
     xarray_open_kwargs = _default_kwargs 
   else:
     xarray_open_kwargs.update(_default_kwargs)
  
```

File: ecgtools/parsers/cesm.py [hunk 2]
```diff
@@ -126,7 +126,7 @@ def parse_cesm_history(file, user_streams_dict={}):
                 except:
                     info['member_id'] = None
                 break
-        with xr.open_dataset(file, chunks={}, decode_times=False) as ds:
+        with xr.open_dataset(file, chunks={}, engine=xarray_open_engine, decode_times=False) as ds:
```
Comment (andersy005) [inline ecgtools/parsers/cesm.py:129]:
```suggestion
        with xr.open_dataset(file, **xarray_open_kwargs) as ds:
```

Comment (andersy005) [reply ecgtools/parsers/cesm.py:129]: What do you think?

Comment (mgrover1) [reply ecgtools/parsers/cesm.py:135]: This makes sense! Seems cleaner...

**Review by andersy005 — APPROVED** [review]

**Pull request merged**